src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -35,6 +39,22 @@ ProvinceInstance::ProvinceInstance(
 	religion_distribution {},
 	max_supported_regiments { 0 } {}
 
+GoodDefinition const* ProvinceInstance::get_rgo_good() const {
+	if(!rgo.is_valid()) { return nullptr; }
+	return &(rgo.get_production_type()->get_output_good());
+}
+void ProvinceInstance::set_rgo_production_type(ProductionType const* rgo_production_type_nullable) {
+	if(rgo_production_type_nullable != nullptr) {
+		ProductionType const& rgo_production_type = *rgo_production_type_nullable;
+		if(rgo_production_type.get_template_type() != ProductionType::template_type_t::RGO) {
+			//error
```

Comment:
Make set_rgo_production_type return a bool which is false if this error occurs and true otherwise and pass it up via whatever functions call this, and Logger::error a message here too.